[MLOB-4262]Add llmobs cost tracking documentation (#32966)
DataDog/documentation: content/en/llm_observability/instrumentation/sdk.md

This gives you the flexibility to either rely on automatic version management based on template content changes, or maintain full control over versioning with your own version labels.

## Cost tracking

> Attach token or cost metrics to the LLM/embedding span so you can view cost data.

Reviewer: ```suggestion
Attach token metrics (for automatic cost tracking) or cost metrics (for manual cost tracking) to your LLM/embedding spans. Token metrics allow Datadog to calculate costs using provider pricing, while cost metrics let you supply your own pricing when using custom or unsupported models.
```
Maybe let's add a link to the cost tracking page here as well

{{< tabs >}}
{{% tab "Python" %}}
Use `LLMObs.annotate(metrics=...)` to attach cost metrics for a LLM/embedding call. For more details on span annotation, see [Annotating a span](#annotating-a-span).

#### Arguments

{{% collapse-content title="Arguments" level="h4" expanded=false id="cost-tracking-arguments" %}}

`metrics`

> : required - dictionary

Reviewer: ```suggestion
: optional - dictionary
```

> <br />Token: `input_tokens`, `output_tokens`, `total_tokens`, `time_to_first_token`, `time_per_output_token`

Reviewer: ```suggestion
<br />Token: `input_tokens`, `output_tokens`, `total_tokens`, `non_cached_input_tokens`, `cache_read_input_tokens`, `cache_write_input_tokens`
```
